anta/inventory/__init__.py

```diff
@@ -190,8 +187,9 @@ def _is_device_online(self, device: InventoryDevice, timeout: float = 5) -> bool
             setdefaulttimeout(timeout)
```

Comment:
This line raises a TypeError if `timeout` is not an integer or string (which happened before the use of the `type` arguement in argparse).
If a user does this mistake, I would expect the code to raise an exception.
So I suggest to take the `setdefaulttimeout` function out of the `try` block.
Also, it seems to be tough to cleanly set the timeout for this library (https://stackoverflow.com/questions/372365/set-timeout-for-xmlrpclib-serverproxy), but maybe set back the default timeout to the previous value so it does not affect other sockets opened by the ANTA user would be great (not necessarily `None` like the StackOverflow example but the previous default timeout).